chord_metadata_service/restapi/middleware.py

```diff
@@ -1,86 +1,104 @@
-            # dataset info beforehand so as to be able to include it here
-            # so we know which authz rule to apply
-            # Bit clumsy, consider this a stop-gap measure
-            access_level = self.decode_permissions(request)
-
-            if not access_level:
-                return HttpResponseForbidden()
-
-            try:
-                allowed = self.query_opa(request, access_level)
-            except requests.exceptions.RequestException:
-                return HttpResponseForbidden()
-
-            if not allowed:
-                return HttpResponseForbidden()
-
-        return self.get_response(request)
-
-    def is_applicable_endpoint(self, request):
-        return request.path in APPLICABLE_ENDPOINTS
-
-    def decode_permissions(self, request):
-        if 'X-CanDIG-Dataset' not in request.headers:
-            return
-        else:
-            dataset = request.headers.get('X-CanDIG-Dataset')
-
-        if 'X-CanDIG-Authz' in request.headers:
-            authz_jwt = request.headers.get("X-CanDIG-Authz").split(' ')[1]
-            # TODO: yes it is a terrible idea to decode without checking the
-            # the signature, gotta find a proper way to transmit vault's public key
-            decoded_jwt = jwt.decode(authz_jwt, verify=False)
-
-            if 'permissions' in decoded_jwt:
-                permissions = decoded_jwt.get('permissions')
-
-                # TODO: I am assuming the permission object is a simple dict such as
-                # {"DATASET_NAME": 4}
-                # TODO: would make sense to package this feature, the reading of the
-                # permission object in a lib, since we'll repeat that stuff over many services
-                if dataset in permissions:
-                    return permissions[dataset]
-                else:
-                    return
-
-    def query_opa(self, request, access_level):
-        if not settings.CANDIG_OPA_URL:
-            return False
-
-        params = {
-            "input": {
-                "path": request.path,
-                "method": request.method,
-                "access_level": int(access_level)
+        """
+        You may incorporate any data source for dataset authorization.
+
+        Note, if no datasets are authorized, you MUST set it to 'NO_DATASETS_AUTHORIZED'.
+
+        If authorized datasets exist, set it to a comma-separated string with titles of authorized datasets.
+
+        For example, if datasets d100 and d200 are authorized, you should set
+        self.authorize_datasets = 'd100,d200'
+        """
+        
+        if settings.CANDIG_AUTHORIZATION == 'OPA' and request.method == 'GET'\
+            and any(re.match(path_re, request.path) for path_re in self.authorized_paths):
+            if settings.CACHE_TIME != 0:
+                error_response = {
+                    "error": "cache time needs to be zero to be secure"
```

Comment:
I think it might be better to say, `This request failed because caching is not disabled. Please contact your system administrator for assistance.` 

Instead of a 500, which is `HttpResponseServerError` below, I think it's better to return a 403, which is `HttpResponseForbidden`.